### PR TITLE
Fix bad snapshot error that was caused by archer changing class on tent or other respawn points

### PR DESCRIPTION
**Pull request by GoldenGuy**
## Status

- **READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description

When player changes class from archer to archer in tent or any other respawn, there might be a bad sync, that would cause a bad snapshot, or sometimes bad delta spam that would crash the server. Here i replaced Sync with Command to prevent this from happening.

## Steps to Test or Reproduce

Quickly press E and pick archer class couple of times, you will see errors in console.

### CODE REVIEW

File: Entities/Characters/Archer/ArcherLogic.as [hunk 1]
```diff
@@ -342,7 +343,12 @@ void ManageBow(CBlob@ this, ArcherInfo@ archer, RunnerMoveVars@ moveVars)
 		if (hasarrow != this.get_bool("has_arrow"))
 		{
 			this.set_bool("has_arrow", hasarrow);
-			this.Sync("has_arrow", isServer());
+			CBitStream params;
+			params.write_bool(hasarrow);
+			if(isServer())
```
Comment (laws65) [inline Entities/Characters/Archer/ArcherLogic.as:348]: Add a space between if and (

File: Entities/Characters/Archer/ArcherLogic.as [hunk 2]
```diff
@@ -429,7 +435,12 @@ void ManageBow(CBlob@ this, ArcherInfo@ archer, RunnerMoveVars@ moveVars)
 			if (responsible)
 			{
 				this.set_bool("has_arrow", hasarrow);
-				this.Sync("has_arrow", isServer());
+				CBitStream params;
+				params.write_bool(hasarrow);
+				if(isServer())
```
Comment (laws65) [inline Entities/Characters/Archer/ArcherLogic.as:440]: Same here

**Review by laws65 — CHANGES_REQUESTED** [review]

**Comment by GoldenGuy** [comment]
@laws65 here

File: Entities/Characters/Archer/ArcherLogic.as [hunk 3]
```diff
@@ -1061,6 +1072,14 @@ void onCommand(CBlob@ this, u8 cmd, CBitStream @params)
 			CycleToArrowType(this, archer, type);
 		}
 	}
+	else if (cmd == this.getCommandID("has arrows"))
+	{
+		bool hasarrow;
+		if(params.saferead_bool(hasarrow))
```
Comment (laws65) [inline Entities/Characters/Archer/ArcherLogic.as:1078]: Also here, sorry I missed it before

File: Entities/Characters/Archer/ArcherLogic.as [hunk 4]
```diff
@@ -342,7 +343,12 @@ void ManageBow(CBlob@ this, ArcherInfo@ archer, RunnerMoveVars@ moveVars)
 		if (hasarrow != this.get_bool("has_arrow"))
 		{
 			this.set_bool("has_arrow", hasarrow);
-			this.Sync("has_arrow", isServer());
+			CBitStream params;
+			params.write_bool(hasarrow);
+			if (isServer())
+				this.SendCommand(this.getCommandID("has arrows"), params);
+			else
+				this.SendCommandOnlyServer(this.getCommandID("has arrows"), params);
```
Comment (eps0003) [inline Entities/Characters/Archer/ArcherLogic.as:351]: I haven't tested this but couldn't these lines be simplified to something like this?
```angelscript
this.SendCommand(this.getCommandID("has arrows"), params, isServer());
```

Comment (GoldenGuy) [reply Entities/Characters/Archer/ArcherLogic.as:351]: only CRules SendCommand has this bool at the end, CBlob function doesnt have it

**Pull request closed without merging**